ironfish/src/wallet/recentFeeCache.ts

```diff
@@ -0,0 +1,132 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { Assert } from '../assert'
+import { Blockchain } from '../blockchain'
+import { createRootLogger, Logger } from '../logger'
+import { PriorityQueue } from '../memPool'
+import { MetricsMonitor } from '../metrics'
+import { Block, Transaction } from '../primitives'
+import { Queue } from './queue'
+
+export class RecentFeeCache {
+  private queue: Queue<Transaction>
+  readonly chain: Blockchain
+  private readonly logger: Logger
+  private readonly metrics: MetricsMonitor
+  private numOfRecentBlocks = 10
+  private numOfTxSamples = 3
+  private defaultSuggestedFee = BigInt(2)
+
+  constructor(
+    chain: Blockchain,
+    recentBlocksNum?: number,
+    txSampleSize?: number,
+    metrics?: MetricsMonitor,
+    logger?: Logger,
```

Comment:
We can use objects here to match our other top level items, because it gets bad when you need to expand this later and you have optionals.

Reply:
use options as input